plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMetadataPrivilegesRequired.java

```diff
@@ -0,0 +1,173 @@
+
+public class TestMetadataPrivilegesRequired
+{
+    private static final String ROOT_USER = "root";
+    private static final String ROOT_PASSWORD = "password";
+    private static final String TEST_USER = "testUser";
+    private static final String TEST_PASSWORD = "pass";
+    private static final String TEST_ROLE = "testRole";
+    private static final String TEST_DATABASE = "test";
+    private static final String TEST_COLLECTION = "testCollection";
+
+    private final AutoCloseableCloser afterClassCloser = AutoCloseableCloser.create();
+
+    private QueryRunner queryRunnerWithDefaultSettings;
+    private QueryRunner queryRunner;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        MongoServer mongo = afterClassCloser.register(setupMongoServer());
+        Map<String, String> defaultMongoProperties = ImmutableMap.of(
+                "mongodb.case-insensitive-name-matching", "true",
+                "mongodb.connection-url", mongo.getConnectionString(TEST_USER, TEST_PASSWORD).toString() + "/?authSource=" + TEST_DATABASE);
+        queryRunnerWithDefaultSettings = afterClassCloser.register(createQueryRunner(defaultMongoProperties));
+        queryRunner = afterClassCloser.register(createQueryRunner(ImmutableMap.<String, String>builder()
+                .putAll(defaultMongoProperties)
+                .put("mongodb.metadata-privileges-required", "false")
+                .buildOrThrow()));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        try (afterClassCloser) {
+            //
+        }
+        finally {
+            queryRunnerWithDefaultSettings = null;
+            queryRunner = null;
+        }
+    }
+
+    @Test
+    public void testSchemasVisibilityWithDefaultConfiguration()
+    {
+        assertThatThrownBy(() -> queryRunnerWithDefaultSettings.execute("SHOW SCHEMAS FROM mongodb"))
+                .hasMessageStartingWith("Command failed with error 13 (Unauthorized)");
+    }
+
+    @Test
+    public void testTablesVisibilityWithDefaultConfiguration()
+    {
+        assertThatThrownBy(() -> queryRunnerWithDefaultSettings.execute("SHOW TABLES FROM mongodb." + TEST_DATABASE))
+                .hasMessageStartingWith("Command failed with error 13 (Unauthorized)");
+    }
+
+    @Test
+    public void testSchemasVisibility()
+    {
+        assertThat(queryRunner.execute("SHOW SCHEMAS FROM mongodb")
+                .getOnlyColumnAsSet())
+                .containsExactly("information_schema", TEST_DATABASE);
+    }
+
+    @Test
+    public void testTablesVisibility()
+    {
+        assertThat(queryRunner.execute("SHOW TABLES FROM mongodb." + TEST_DATABASE)
+                .getOnlyColumnAsSet())
+                .containsExactly(TEST_COLLECTION.toLowerCase(Locale.ENGLISH));
+    }
+
+    private static MongoServer setupMongoServer()
+    {
+        MongoServer server = new MongoServer("4.0.6", ImmutableMap.of(
```

Comment:
MongoDB 4.0 already reached EOL on Apr 30. Let's use a new version. Upcoming FTE on MongoDB will upgrade the minimum supported version to 4.2.

Reply:
I set version to `4.4.17` - it seams docker init flakiness (when I set root user and password via env) got fixed

Reply:
Hmm, it's still failing in my environment.

Reply:
Looks like `org.testcontainers.containers.MongoDBContainer` contains a bug.

When I set `MONGO_INITDB_ROOT_USERNAME` and `MONGO_INITDB_ROOT_PASSWORD` this cause a restart MongoDB process.
At the same time `db.runCommand( { isMaster: 1 } ).ismaster==false` is executed inside Docker container in a loop (60 times) and during process restart this throw network error - that's the main issue.

I create `AuthenticatedMongoServer` which use `GenericContainer` directly. Default enabled `replication` is not needed in current scenario.